gef-ui/server/controllers/supporting-information/manual-inclusion-questionnaire/index.js

```diff
@@ -0,0 +1,237 @@
+const Application = require('../../../models/application');
+const { validationErrorHandler } = require('../../../utils/helpers');
+const validateFile = require('../../../utils/validateFile');
+const { uploadAndSaveToDeal, removeFileFromDeal } = require('../../../utils/fileUtils');
+
+// This is only needed in this question due to the different size requirement
+const MAX_FILE_SIZE = 1024 * 1024 * 12;
+const FIELD_NAME = 'manualInclusion';
+
+const getUploadManualInclusion = async (req, res) => {
+  const {
+    session: { user, userToken },
+    params: { applicationId },
+  } = req;
+
+  const application = await Application.findById(applicationId, user, userToken);
+
+  if (!application) {
+    console.log(`User unauthorised to view application ${applicationId} manual inclusion`);
+    return res.redirect('/dashboard/gef/');
+  }
+
+  return res.render('partials/manual-inclusion-questionnaire.njk', {
+    user,
+    applicationId,
+    files: application.supportingInformation?.manualInclusion,
+  });
+};
+
+const validateFileQuestion = (application, field, errRef) => {
+  const errors = [];
+
+  const files = application?.supportingInformation?.[field] || [];
+
+  if (!files.length) {
+    errors.push({
+      errRef,
+      errMsg: 'Upload Manual inclusion questionnaire',
+    });
+  }
+
+  if (files.length > 20) {
+    errors.push({
+      errRef,
+      errMsg: 'You can only upload up to 20 files at the same time',
+    });
+  }
+
+  return errors;
+};
+
+const postUploadManualInclusion = async (req, res) => {
+  const {
+    body: { delete: fileToDelete, submit },
+    files,
+    params: { applicationId },
+    session: { user, userToken },
+  } = req;
+  const errRef = 'documents';
+
+  let manualInclusionErrors = [];
+  let processedFiles = [];
+
+  let application = await Application.findById(applicationId, user, userToken);
+
+  if (!application) {
+    console.log(`User unauthorised to view application ${applicationId} manual inclusion`);
+    return res.redirect('/dashboard/gef/');
+  }
+
+  console.log('uploading file');
```

Comment:
remove noise?

Reply:
good catch - one last bit of debugging I left